cassandra/src/main/java/se/tre/freki/storage/cassandra/CassandraStore.java

```diff
@@ -384,18 +391,118 @@ public Boolean apply(final ResultSet update) {
 
   @Nonnull
   @Override
-  public ListenableFuture<LabelMeta> getMeta(final LabelId uid,
-                                             final LabelType type) {
-    throw new UnsupportedOperationException("Not implemented yet");
+  public ListenableFuture<Optional<LabelMeta>> getMeta(final LabelId id,
+                                                       final LabelType type) {
+
+    final ListenableFuture<Optional<Row>> metas = getOptionalMeta(id, type);
+    return transform(metas, new Function<Optional<Row>, Optional<LabelMeta>>() {
+      @Nullable
+      @Override
+      public Optional<LabelMeta> apply(final Optional<Row> result) {
+        if (!result.isPresent()) {
+          return Optional.absent();
+        }
+
+        final Row row = result.get();
+
+        return Optional.of(
+            LabelMeta.create(CassandraLabelId.fromLong(row.getLong("label_id")),
+                LabelType.fromValue(row.getString("type")), row.getString("name"),
+                row.getString("meta_description"),
+                row.getDate("creation_time").getTime()));
+      }
+    });
+
+  }
+
+  @Nonnull
+  ListenableFuture<Optional<Row>> getOptionalMeta(final LabelId id,
+                                                  final LabelType type) {
+
+    final ListenableFuture<List<Row>> rowFuture = fetchLabelRows(id, type,
+        getMetaStatement.bind(toLong(id), type.toValue()));
+
+    return transform(rowFuture, new FirstOrAbsentFunction<Row>());
+  }
+
+  /**
+   * Use this function to check for duplicates.
```

Comment:
Executes the provided statement and logs if the result contains more than one statement. Due to the logging message it is expected that the statement will fetch a an ID.